src/ShallowTestWrapper.js

```diff
@@ -17,11 +19,13 @@ export default class ShallowTestWrapper extends TestWrapper {
   }
 
   inspect () {
-    const name = this.wrapper.root.unrendered.type.displayName ||
-      this.wrapper.root.unrendered.type.name ||
-      '???'
+    const root = this.wrapper.root()
+
+    const rootInstance = root.instance()
+    const rootType = rootInstance ? rootInstance.constructor : root.getElement().type
+    const name = rootType ? getDisplayName(rootType) : '???'
```

Comment:
`root.name() || '???'` should suffice here, no?

Reply:
Enzyme's `ShallowWrapper` behaves slightly different from the `ReactWrapper` when it comes to the `name()` function. Given the following component:
```js
const Example = () => <table />;
```

`mount(<Example />).name()` returns `'Example'` while `shallow(<Example />).name()` returns `'table'`.